src/shogun/util/factory.h

```diff
@@ -333,5 +335,63 @@ namespace shogun
 	{
 		return std::make_shared<PipelineBuilder>();
 	}
+
+	class StringsFeatures;
+	class FeatureSubset;
+
+	template <
+	    typename TypeName, typename... Args,
+	    std::enable_if_t<
+	        traits::is_any_of_v<
+	            TypeName, Features, StringsFeatures, FeatureSubset,
```

Comment:
question: can we avoid having to write these class names in-place here, rather than putting them into a list somewhere else, together with other base type definitions?

Reply:
there was an initiative to put the base interface classes into base_types.h see sg_inferface

Reply:
you can also the overload the other way around. Specialise for string types, and just have this without any SFINAE. Should be easier to maintain.. And then have a static_assert at the end of the constexpr that throws a compile time error if it cannot handle. Downside is that you need to handle, const char*, std::string and std::string_view, or check if the type is implicitly convertible to string-like...